billchop-fe/src/components/SearchBox.tsx

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import { debounce } from "lodash";
+import { Button, Col, FormControl, InputGroup, ListGroup, Row } from "react-bootstrap";
+import SearchIcon from "../assets/search-icon.svg";
+import ArrowBackIcon from "../assets/arrow-back-icon.svg";
+import OutsideClickListener from "./OutsideClickListener";
+import "../styles/search-box.css";
+
+export interface ISearchBoxProps {
+  placeholder?: string;
+  searchResults: Map<string, string>; //key - id, value - search result item
+  onChange: (keyword: string) => void;
+  onActionButtonClick: (selectedItemId: string) => void;
+  actionButtonText: string;
+  onHide: () => void;
+}
+
+interface ISearchBoxState {
+  inputValue: string;
+}
+
+export default class SearchBox extends React.Component<
+  ISearchBoxProps,
+  ISearchBoxState
+  > {
+  constructor(props: ISearchBoxProps) {
+    super(props);
+
+    this.state = {
+      inputValue: "",
+    };
+  }
+
+  handleInputChange = (inputValue: string): void => {
+    if (!inputValue) return;
+
+    const { onChange } = this.props;
+    onChange(inputValue);
+  };
+
+  delayedHandleInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const inputValue = event.target.value;
+    this.setState({ inputValue });
+    const delayed = debounce(() => this.handleInputChange(inputValue), 300);
+    delayed();
```

Comment:
Does this work? I'm not 100% familiar with how debounced works, perhaps it has some global state, but to me this looks like you are creating a new debounced version of the function every time, which means it's probably not debounced. Try increasing the time to something bigger (10s?) and see if it works.

To make debounced work, I think you will need to do something like this:
```typescript
  handleInputChange = (inputValue: string): void => {
    if (!inputValue) return;

    const { onChange } = this.props;
    onChange(inputValue);
  };

  handleDebouncedInputChange = debounce(this.handleInputChange, 300);

  delayedHandleInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
    const inputValue = event.target.value;
    this.setState({ inputValue });

    this.handleDebouncedInputChange(inputValue);
  };
```